src/builder.ts

```diff
@@ -170,24 +172,41 @@ export const UNKNOWN_TYPE_SCALAR = decorateType(
   }
 );
 
-export interface BuilderConfig {
+export interface BuilderOptions {
   /**
-   * When the schema starts and `process.env.NODE_ENV !== "production"`,
-   * artifact files are auto-generated containing the .graphql definitions of
-   * the schema
+   * Generated artifact settings. Set to false to disable all.
+   * Set to true to enable all and use default paths. Leave
+   * undefined for default behaviour of each artifact.
    */
-  outputs:
+  outputs?:
+    | boolean
     | {
         /**
-         * Absolute path where the GraphQL IDL file should be written
+         * TypeScript declaration file generation settings. This file
+         * contains types reflected off your source code. It is how
+         * Nexus imbues dynamic code with static guarnatees.
+         *
+         * Defaults to being enabled when `process.env.NODE_ENV !== "production"`.
+         * Set to true to enable and emit into default path (see below).
+         * Set to false to disable. Set to a string to specify absolute path.
+         *
+         * The default path is node_modules/@types/__nexus-typegen__core/index.d.ts.
```

Comment:
I recently did a deep-dive in the rules of type resolution but can't remember now, whether TS picks up `.ts` as well as `.d.ts` files by default in `@types`. 

I tend to use `.ts` files for the nexus gen in my own projects, because I often have `"skipLibCheck": true,` set in `tsconfig` (to suppress extraneous errors in 3rd party dependencies). When this is set, any type errors in all `.d.ts` files are silently ignored, which can be a gotcha.

Reply:
> When this is set, any type errors in all .d.ts files are silently ignored, which can be a gotcha.

I just tried `skipLibCheck` on the nexus-prisma example app (which generates into `@types`) and got this result: 

![image](https://user-images.githubusercontent.com/284476/65799106-4ea3eb00-e141-11e9-8059-4281844be75a.png)

Did I run my test incorrectly? Or did I misunderstand your point?

Reply:
Point was if there was a broken import or custom type or something in the `.d.ts`, I believe it would just turn the error into an `any` rather than actually reporting the error.

Reply:
Ah gotchya. And yeah, catching up to you here, confirmed.

If I attempt to use `index.ts` instead of `index.d.ts` in `@types` packages I get a compile error `error TS2688: Cannot find type definition file for '__nexus-typegen__nexus-core'.`.

I will do a bit of research to see if there is any solution that would permits the benefits of using `@types` without succumbing to the effects of `skipLibCheck`.

Reply:
So did not find a solution; thoughts:

- A baseline point, obviously it is not ideal to use `skipLibCheck`. Two use-cases originally IIUC were 1) work around incompatible dep/transient dep typing integration issues; 2) compile perf. With modern TS incremental build stuff, I would hope perf use-case is gone/going.
- IIUC/Question: it should be impossible for user code to introduce bug in typegen
- When user _needs_ `skipLibCheck`, they still benefit from typegen in `@types`, so long as the typegen is bug free
- Considering the above points in sum, IIUC, using `@types` as default is problematic only when the stars align thusly: 1) nexus lib has a typegen bug 2) user hits the bug path 3) user _needs_ `skipLibCheck`.
- Escape hatch solution: user manually configures typegen paths

I think such a tradeoff would be reasonable.

Reply:
I found a discussion for a granular version of `skipLibCheck` here https://github.com/microsoft/TypeScript/issues/30511. If TS had this feature, it would further reduce pressure against nexus type in `@types` being a good default, because some amount of current `skipLibCheck` cases would be able to adjust to the granular form.

I will add our use-case to the discussion over there.

Reply:
> Point was if there was a broken import or custom type or something in the .d.ts, I believe it would just turn the error into an any rather than actually reporting the error.

Would it still be the case with `noImplicitAny: true` ?

I generally agree with your point @jasonkuhrt. The whole point of such defaults is to reduce friction for newcomers. I'm still a bit concerned by the issue cited above though. Broken import can happen easily when messing with `typegenAutoConfig`, and inputting some wrong file paths

Reply:
> noImplicitAny: true

Yep still the same.

> Broken import can happen easily when messing with typegenAutoConfig (to enhance the backing types), and inputting some wrong file paths

Ah, thanks for clarifying that. Agree that increases the downside potential.

Still feel it is the right tradeoff.

We probably want good user docs for this part of the system. User awareness can do its part to help narrow the unhappy path.

Reply:
Yeah I think what you have as the default is reasonable. Let's do it.